src/Services/FHIR/FhirAllergyIntoleranceService.php

```diff
@@ -52,6 +56,60 @@ protected function loadSearchParameters()
         ];
     }
 
+
+    /**
+     * Parses an OpenEMR allergyIntolerance record, returning the equivalent FHIR AllergyIntolerance Resource
+     *
+     * @param array $dataRecord The source OpenEMR data record
+     * @param boolean $encode Indicates if the returned resource is encoded into a string. Defaults to false.
+     * @return FHIRAllergyIntolerance
+     */
+    public function createProvenanceResource($dataRecord = array(), $encode = false)
+    {
+        
+        $allergyProvenance = new FHIRProvenance();
+        $meta = array('versionId' => '1', 'lastUpdated' => gmdate('c'));
+        $allergyProvenance->setMeta($meta);
+        
+        $id = new FHIRId();
+        $uuidString = UuidRegistry::uuidToString((new UuidRegistry(['disable_tracker' => true]))->createUuid());
+        $id->setValue($uuidString);
+        error_log("uuid is " . $uuidString);
```

Comment:
hi @vishnuyar ,
will need to remove this error_log

also, what endpoint/path do you use to request the provenance in the fhir api call?

Reply:
Removed the same. 
As mentioned in this https://build.fhir.org/ig/HL7/US-Core-R4/basic-provenance.html
The API call for Allergy provenance request will be 
GET /AllergyIntolerance?patient=[ID]&_revinclude=Provenance:target